### ISSUE TITLE
Aria-live regions not read correctly

### ISSUE THREAD

**Issue by miksa1987** [nvaccess/nvda#11408]
Hi, hopefully someone here has some time to look into my issue and this is the right channel.. Thank you in advance.

I'm in the process of making our company's React app as accessible as possible. However, I'm noticing some weird behavior with NVDA - I have two live regions which I just cannot get NVDA to announce correctly. On Voiceover, everything works smoothly with these.

First there is a notification component, which has code similar to this. Like this, NVDA doesn't announce the notifications, Voiceover does:

< div aria-live="polite" aria-hidden="true/false">
  < article> <-- There might be many of these if there's multiple notifications
    < div>
      < div type="alert">
        < div>
          < div>The text here should be announced when shown</div>
        < /div>
      < /div>
    < /div>
  < /article>
< /div>

If I remove aria-hidden from the first div, NVDA reads it, but it gets unreliable on Voiceover. Inside the first div, there is a PoseGroup component too, just doesn't show up on generated HTML.

Then there's a live chat, which has code like this:

< div>
  < ul aria-live="polite"> <-- There are multiple of these depending on number of elements
    < li aria-hidden="true/false"> <--- There are many of these, some hidden from screen reader depending on the component, these are messages, timestamps etc, some of them should be read
      < div>
        < span>
          < div>
            < p aria-hidden="true">Visible message text< /p>
            < p aria-hidden="false">Hidden, but contains additional information to screen reader and should be read always< /p>
         < /div>
      < /span>
      < /div>
    < /li>
  < /ul>
< /div>

Here, I'm using multiple aria-live containers which are spawned when there are enough messages on the former one, this is due to Voiceover stopping the reading when there are too many elements, and this solution works with it. It should be also noted that there is a PoseGroup component wrapping the li elements, it doesn't show on the generated HTML. The li elements should only read the messages, no timestamps etc, that's why they have aria-hidden. No matter what I tried, NVDA doesn't read any of these. Voiceover works perfectly with this kind of code. I have tried to convert the ul and li's to divs with no luck.

Then there's a form input component, which has an error label as a live region, goes like this:

< label>
  < span>Some text</span>
  < input/>
  < div aria-live="polite">
    < span>This span is conditionally shown if there's an error.</span>
  < /div>
< /label>

This component however, works perfectly like this on both Voiceover and NVDA, and I just cannot get my head around why the other two don't. Sure, they're more complex, but shouldn't it work on NVDA too since Voiceover does it without any difficulty? 

What I'm asking here is a bit of assistance, if anyone would have an idea of how I could get both screen readers to work? I still consider myself as an accessibility noob, and been banging my head against a wall for couple of days.

I'm using latest NVDA on latest update of Windows 10 for testing this, on an HP EliteBook with 8GB RAM and an i7-6500U CPU.

EDIT: Sorry about bad formatting - I'm really bad at it. Hope you get the idea.

**Comment by feerrenrut** [comment]
Please fill out the issue template. In particular the "expected" and "actual" behaviour (hint, use speech viewer from the tools menu to capture this). From the current issue description I can't determine what problem you are encountering. I'd suggest first starting with one example, and clearly explain what you are trying to achieve, and the unexpected behavior you are encountering instead. It is useful to describe how you are testing this (exact steps, eg key presses). This helps us to help you. Until this is done, I'm going to close this issue.

Both of the following resources are classic guides to make this process more effective:
- [How To Ask Questions The Smart Way, by Eric Raymond](http://catb.org/~esr/faqs/smart-questions.html)
- [How to Report Bugs Effectively, by Simon Tatham](https://www.chiark.greenend.org.uk/~sgtatham/bugs.html)
